react-redux/react-redux.d.ts

```diff
@@ -99,5 +103,9 @@ declare module "react-redux" {
   /**
    * Makes the Redux store available to the connect() calls in the component hierarchy below.
    */
-  export class Provider extends Component<ProviderProps, {}> { }
+  export class Provider extends __React.Component<ProviderProps, {}> { }
+}
+
+declare module "react-redux" {
+  export = ReactRedux;
```

Comment:
Tests pass and this is the same pattern used by Redux and others, so AFAICT this should be a safe change.